#**PRACTICAL** 1
**AIM:** Identify and classify different data types and compute basic statistical measures to summarize dataset characteristics

In [11]:
import pandas as pd
import json
from sklearn.datasets import load_iris
from collections import Counter

I/O-related

In [12]:
# Import Iris dataset using Pandas
iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)

df['species'] = iris.target
df['species'] = df['species'].map(dict(enumerate(iris.target_names)))
#display first 5 rows
print("FIRST 5 ROWS OF DATASET:\n")
print(df.head())
#display last 5 rows
print("\nLAST 5 ROWS OF DATASET:\n")
print(df.tail())
#display shape column names, and data types of the dataset
print("\nSHAPE OF DATASET:", df.shape)
print("\nCOLUMN NAMES:\n", df.columns.tolist())
print("\nDATA TYPES:\n", df.dtypes)

# (+) Dataset size information
print("\nTOTAL ROWS:", df.shape[0])
print("TOTAL COLUMNS:", df.shape[1])

# (+) Missing value check
print("\nMISSING VALUES PER COLUMN:\n")
print(df.isnull().sum())



FIRST 5 ROWS OF DATASET:

   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

  species  
0  setosa  
1  setosa  
2  setosa  
3  setosa  
4  setosa  

LAST 5 ROWS OF DATASET:

     sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
145                6.7               3.0                5.2               2.3   
146                6.3               2.5                5.0               1.9   
147                6.5               3.0                5.2               2.0   
148                6.2               3.4                5.4               2.3

Identification of Dataset

In [13]:

# Numerical columns
numerical_cols = df.select_dtypes(include=['int64', 'float64']).columns

# Categorical columns
categorical_cols = df.select_dtypes(include=['object']).columns

ordinal_cols = []

print("\nNUMERICAL COLUMNS:")
print(list(numerical_cols))

print("\nCATEGORICAL COLUMNS:")
print(list(categorical_cols))

print("\nORDINAL COLUMNS:")
print(ordinal_cols if ordinal_cols else "None")

# (+) Unique value count for categorical attributes
print("\nUNIQUE VALUES IN CATEGORICAL COLUMNS:\n")
for col in categorical_cols:
    print(col, ":", df[col].unique())

print("\nSTATISTICAL SUMMARY OF NUMERICAL DATA:\n")
print(df[numerical_cols].describe())

# (+) Additional
print("\nMEAN VALUES:\n")
print(df[numerical_cols].mean())

print("\nMEDIAN VALUES:\n")
print(df[numerical_cols].median())

print("\nSTANDARD DEVIATION:\n")
print(df[numerical_cols].std())



NUMERICAL COLUMNS:
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']

CATEGORICAL COLUMNS:
['species']

ORDINAL COLUMNS:
None

UNIQUE VALUES IN CATEGORICAL COLUMNS:

species : [np.str_('setosa') np.str_('versicolor') np.str_('virginica')]

STATISTICAL SUMMARY OF NUMERICAL DATA:

       sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)  
count        150.000000  
mean           1.199333  
std            0.76223

STRUCTURED VS UNSTRUCTURED DATA

In [14]:
# ----- Structured Data (CSV) -----
print("\nSTRUCTURED DATA (CSV) SUMMARY:\n")
print(df.describe())

# Save CSV file (optional)
df.to_csv("iris_dataset.csv", index=False)

# ----- Unstructured Data (Text File) -----
text_data = """
Iris dataset is a famous dataset used for machine learning.
It contains sepal length, sepal width, petal length, and petal width.
The dataset has three species: setosa, versicolor, virginica.
"""

# Save text file
with open("iris_text.txt", "w") as file:
    file.write(text_data)

# Read text file
with open("iris_text.txt", "r") as file:
    content = file.read()

# Word frequency
words = content.lower().split()
word_freq = Counter(words)

print("\nWORD FREQUENCY IN TEXT DATA:\n")
print(word_freq)

# (+) Top 5 most frequent words
print("\nTOP 5 MOST FREQUENT WORDS:\n")
print(word_freq.most_common(5))




STRUCTURED DATA (CSV) SUMMARY:

       sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)  
count        150.000000  
mean           1.199333  
std            0.762238  
min            0.100000  
25%            0.300000  
50%            1.300000  
75%            1.800000  
max            2.500000  

WORD FREQUENCY IN TEXT DATA:

Counter({'dataset': 3, 'sepal': 2, 'length,': 2, 'petal': 2, 'iris': 1, 'is': 1, 'a': 1, 'famous': 1, 'used': 1, 'fo

In [15]:
# ----- JSON Data Reading (+) -----
json_data = df.head(5).to_dict(orient="records")

with open("iris_data.json", "w") as file:
    json.dump(json_data, file)

with open("iris_data.json", "r") as file:
    json_loaded = json.load(file)

print("\nJSON DATA SAMPLE:\n")
print(json_loaded)



JSON DATA SAMPLE:

[{'sepal length (cm)': 5.1, 'sepal width (cm)': 3.5, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2, 'species': 'setosa'}, {'sepal length (cm)': 4.9, 'sepal width (cm)': 3.0, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2, 'species': 'setosa'}, {'sepal length (cm)': 4.7, 'sepal width (cm)': 3.2, 'petal length (cm)': 1.3, 'petal width (cm)': 0.2, 'species': 'setosa'}, {'sepal length (cm)': 4.6, 'sepal width (cm)': 3.1, 'petal length (cm)': 1.5, 'petal width (cm)': 0.2, 'species': 'setosa'}, {'sepal length (cm)': 5.0, 'sepal width (cm)': 3.6, 'petal length (cm)': 1.4, 'petal width (cm)': 0.2, 'species': 'setosa'}]


About the Iris Dataset

The Iris dataset is one of the most commonly used datasets for learning data analysis and machine learning. It contains information about different Iris flowers and their physical characteristics. Each row in the dataset represents a single flower, and the measurements include sepal length, sepal width, petal length, and petal width.

Along with these numerical values, the dataset also includes a species column, which tells us which type of Iris flower the sample belongs to—Setosa, Versicolor, or Virginica. The dataset has 150 records in total and is well structured, which makes it easy to understand and work with.


The Python program helped in gaining a clear and structured understanding of the Iris dataset. By loading the dataset into a Pandas DataFrame, it became easy to view the data in a tabular form, which made the dataset more readable and understandable. Displaying the first and last few rows helped in understanding how the data is organized and what each record represents.

Identifying the data types allowed us to distinguish between numerical attributes, such as sepal and petal measurements, and the categorical attribute, which is the species of the flower. This step made it clear how different types of data are handled during analysis. Computing statistical measures like mean, median, and standard deviation helped summarize the dataset and provided insights into the spread and variation of each feature.

The program also helped in understanding the structure of the dataset, including the number of rows, columns, and the absence of missing values. Overall, the Python program made the manual analysis of the Iris dataset more meaningful by validating observations through code and simplifying the process of exploring and summarizing the data.